src/docfx/lib/path/PathUtility.cs

```diff
@@ -85,6 +88,27 @@ public static string NormalizeFolder(string path)
         return str;
     }
 
+    public static string NormalizeFileWithValidation(string path)
+    {
+        path = NormalizeFile(path);
+        var invalidPathChars = new HashSet<char>();
+        if (!string.IsNullOrEmpty(path))
+        {
+            foreach (var c in path)
+            {
+                if (s_invalidPathChars.Contains(c))
+                {
+                    invalidPathChars.Add(c);
+                }
+            }
+        }
+        if (invalidPathChars.Count > 0)
+        {
+            throw new InvalidDataException($"Path {path} contains invalid chars {string.Join(", ", invalidPathChars.Select(c => $"'{c}'"))}.");
```

Comment:
throw `DocfxException` here.
```suggestion
            throw Errors.Json.PathInvalid(invalidPathChars).ToException();
```

Reply:
@yufeih Updated as suggested, could you please help review?